### ISSUE TITLE
Bug: pnpm install fails ERR_PNPM_LINKED_PKG_DIR_NOT_FOUND

### ISSUE THREAD

**Issue by MikeMcC399** [eslint/css#56]
### Environment

ESLint version:  N/A
@eslint/css version: `0.4.0 `
Node version:  `v22.14.0` LTS
pnpm version:  `10.4.1`
Operating System: Ubuntu 24.04.2 LTS

This is a regression from @eslint/css version: `0.2.0` 
Also reproducible on @eslint/css version: `0.3.0` 
Also reproducible with `pnpm@10.5.0`

Installations using Yarn 1 Classic `v1.22.22` are also failing.

### Which language are you using?

N/A

### What did you do?

```shell
corepack enable pnpm
cd $(mktemp -d)
corepack use pnpm@10.4.1
pnpm add @eslint/css@0.4.0
```

### What did you expect to happen?

pnpm should install without error

### What actually happened?

 ERR_PNPM_LINKED_PKG_DIR_NOT_FOUND

```text
$ corepack enable pnpm
cd $(mktemp -d)
corepack use pnpm@10.4.1
pnpm add @eslint/css@0.4.0
Installing pnpm@10.4.1 in the project...

Already up to date
Done in 956ms using pnpm v10.4.1
 ERR_PNPM_LINKED_PKG_DIR_NOT_FOUND  Could not install from "/tmp/tmp.Y6rHpfEfdA/typings/css-tree" as it does not exist.

This error happened while installing the dependencies of @eslint/css@0.4.0
Progress: resolved 1, reused 0, downloaded 0, added 0
```

### Link to Minimal Reproducible Example

See above for short repro steps ⬆ 

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
Thanks for the report. pnpm and Yarn do strange things with dependencies to try to "optimize", which can cause problems. Please try the suggested settings in our README and let me know if that helps:
https://github.com/eslint/eslint?tab=readme-ov-file#pnpm-installation

**Comment by MikeMcC399** [comment]
@nzakas 

> Thanks for the report. pnpm and Yarn do strange things with dependencies to try to "optimize", which can cause problems. Please try the suggested settings in our README and let me know if that helps: https://github.com/eslint/eslint?tab=readme-ov-file#pnpm-installation

Adding the suggestions for .npmrc did not change the outcome

## pnpm repro

```shell
corepack enable pnpm
cd $(mktemp -d)
cat > .npmrc <<EOT
auto-install-peers=true
node-linker=hoisted
EOT
corepack use pnpm@10.4.1
pnpm add @eslint/css@0.4.0
```

**Comment by MikeMcC399** [comment]
For comparison

## Yarn Classic repro

fails:

```shell
corepack enable yarn
cd $(mktemp -d)
corepack use yarn@1
yarn add @eslint/css@0.4.0
```

## Yarn (Modern) repro

no failure:

```shell
corepack enable yarn
cd $(mktemp -d)
corepack use yarn@4
yarn add @eslint/css@0.4.0
```

**Comment by CavaliereDavid** [comment]
I’m also encountering this issue when trying to install `@eslint/css@0.4.0` using `pnpm`. Even after applying the recommended `.npmrc` settings:

```
auto-install-peers=true
node-linker=hoisted
```

the installation still fails with:

```
ERR_PNPM_LINKED_PKG_DIR_NOT_FOUND  Could not install from ".../typings/css-tree" as it does not exist.
```

### **Environment:**

```
OS: macOS
Node.js version: v22.12.0
pnpm version: 10.3.0
```

I also enabled `fnm env --use-on-cd` to allow auto-switching between Node.js versions, but the issue persists.

**Comment by MikeMcC399** [comment]
@nzakas 

As a feasibility exercise I forked the repo and added a workflow 

https://github.com/MikeMcC399/css/blob/prime/.github/workflows/ci-build-all-pm.yml

which builds the package and then installs it on different package managers

https://github.com/MikeMcC399/css/actions/runs/13432882751 is an example run which shows where it fails:

![Image](https://github.com/user-attachments/assets/794b60aa-b6e4-4163-af1a-44214c3c1ca7)

I don't know if this should be part of the regular CI. It would be unusual for npm packages to be tested like this. We should cover this in a separate issue if you want to follow up.

**Comment by nzakas** [comment]
@MikeMcC399 thanks! I think that's something we can consider. I'd say go ahead and send a PR as I think that's helpful.

So looking at this a bit closer, it looks like pnpm interprets `file:` as meaning relative to the current working directory vs. relative to the package's installation directory. It seems like this may be a bug, so opened an issue here:
https://github.com/pnpm/pnpm/issues/9141

It also seems like this was a bug in Yarn as the most recent version doesn't exhibit this behavior. Maybe we can just ask people to upgrade their Yarn if they run into this problem?

**Comment by MikeMcC399** [comment]
@nzakas

> I think that's something we can consider. I'd say go ahead and send a PR as I think that's helpful.

I will wait until there is some feedback from pnpm, as I don't want to submit a PR which will fail in CI.

> So looking at this a bit closer, it looks like pnpm interprets `file:` as meaning relative to the current working directory vs. relative to the package's installation directory. It seems like this may be a bug, so opened an issue here: [pnpm/pnpm#9141](https://github.com/pnpm/pnpm/issues/9141)

I'll monitor this issue.

> It also seems like this was a bug in Yarn as the most recent version doesn't exhibit this behavior. Maybe we can just ask people to upgrade their Yarn if they run into this problem?

- I opened a new issue for Yarn under https://github.com/eslint/css/issues/58. Although the error is related, the discussion is different, especially as Yarn v1 Classic is frozen.

**Comment by nzakas** [comment]
Just as a heads-up: We're going to be creating our own fork of CSSTree that will bundle its own types, so we'll be able to move away from this `file:` approach shortly and that should fix everything on the various package managers.

**Comment by MikeMcC399** [comment]
@nzakas 

> Just as a heads-up: We're going to be creating our own fork of CSSTree that will bundle its own types, so we'll be able to move away from this `file:` approach shortly and that should fix everything on the various package managers.

That sounds like a good decision! You can ignore my comment in the Yarn v1 issue in that case 🙂

**Comment by MikeMcC399** [comment]
- PR #83 resolved this issue
- PR #84 adds back the pm check for pnpm
